xsuite/src/proxy/proxy.ts

```diff
@@ -29,22 +29,38 @@ export class Proxy {
     this.blockNonce = params.blockNonce ?? null;
   }
 
-  fetchRaw(path: string, data?: any) {
-    const baseUrl = this.proxyUrl;
-    const init: RequestInit = { headers: this.headers };
-    const url = new URL(`${baseUrl}${path}`);
-    if (this.blockNonce !== null) {
-      url.searchParams.append("blockNonce", this.blockNonce.toString());
+  private makeUrl(
+    path: string | URL,
```

Comment:
Is there a need for URL?

Reply:
Because `makeUrl` returns an `URL` and in the current state we can call `makeUrl` 2 times in a row for the same request, so I think its better to stay with only one `URL` object instead of creating an `URL`, `.toString()` the URL, re-creating an `URL`, ...

Reply:
I see :+1: Let's not resolve and let's wait for the other points to be resolved before concluding on this one.